utilities/mobile_server_pool.py

```diff
@@ -34,15 +36,27 @@ def get_nodes_available_from_mobile_pool(nodes_os_type, node_os_version):
     :return: list of nodes reserved
     """
     # Getting list of available nodes through n1ql query
-    query_str = "select count(*) from `{}` where os='{}' " \
-                "AND os_version='{}' AND state='available'".format(BUCKET_NAME, nodes_os_type, node_os_version)
+    device_slave_ip = ""
+    if nodes_os_type in MOBILE_OS:
+        device_slave_ip = " AND slave_ip = '{}'".format(slave_ip)
+    query_str = "select  meta().id from `{}` where os='{}' " \
+                "AND os_version='{}' AND state='available'"\
+                " {}".format(BUCKET_NAME, nodes_os_type, node_os_version, device_slave_ip)
     query = cluster.query(query_str)
+    count = 0
     for row in query:
-        count = row["$1"]
+        doc_id = row["id"]
+        if nodes_os_type in MOBILE_OS:
+            check_alive = check_device_alive(doc_id)
+        else:
+            check_alive = check_vm_alive(doc_id)
+        if check_alive:
+            count += 1
     print(count)
```

Comment:
Do you return the list of IPs? I see a print of the count, but no value returned

Reply:
No, it returns only number of available vm and could remote access to it.